### PR TITLE
Initial configurable PSI deployment

### PR DESCRIPTION
**Pull request by jlhcrawford**
Initial commit for the configurable PSI deployment.
Includes
- source code in `deployments/config_psi/psi`
- Gtest unit tests in `deployments/config_psi/psi/tests`
- python scripts for encode and decode in `deployments/config_psi/scripts`
- pytests for scripts in `deployments/config_psi/scripts/tests`
- pytests for `deployments/config_psi/tests`
- Initial version of configurable PSI README (`deployments/config_psi/README.md`).
- Recipe file for building configurable PSI deployment and required dependencies `recipes/config-psi.toml`

### CODE REVIEW

File: deployments/config_psi/README.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+HElib >= v2.2.1
+HElib utils (for encrypt and decrypt)
+```
+The easiest method for installing the required dependencies of this deployment
+is to use `hekit` to install the `recipes/config-psi.toml` recipe file.
+
+
+## Usage
+The `config_psi` directory is split into `psi` and `scripts`. The `psi`
+directory contains code which performs the core PSI operation. The `scrips`
```
Comment (hamishun) [inline deployments/config_psi/README.md:23]:
```suggestion
directory contains code which performs the core PSI operation. The `scripts`
```

File: deployments/config_psi/README.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+```
+The first row is the column headings, and the subsequent rows are the input
+data. These columns can have alphanumeric, alphabetical, and numeric values.
+
+### Encode
+To encode data like the example shown above, users can use the script provided
+in `scripts/encode.py`.
+
+This script performs two types of encoding; client
+encoding (default), and server encoding when supplied the `--server` flag.
```
Comment (hamishun) [inline deployments/config_psi/README.md:49]:
```suggestion
encoding (default) and server encoding when supplied the `--server` flag.
```

File: deployments/config_psi/README.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+[here](https://github.com/homenc/HElib/tree/master/utils).
+
+**NOTE:** The `create-context` utilities is required to produce the necessary
+encryption/decryption keys.
+
+### PSI
+The PSI deployment performs the PSI computation between two
+plaintext/ciphertext datasets using a custom query. The query can be passed via
+what we call a `table file`. This table file has the following format
+```
```
Comment (hamishun) [inline deployments/config_psi/README.md:83]: Is TOML supported here. Github markdown may not at the mo, but worth checking.

File: deployments/config_psi/psi/tests/main.cpp [hunk 4]
```diff
@@ -0,0 +1,17 @@
+// *****************************************************************************
+// INTEL CONFIDENTIAL
+// Copyright 2021 Intel Corporation
+//
+// This software and the related documents are Intel copyrighted materials, and
+// your use of them is governed by the express license under which they were
+// provided to you ("License"). Unless the License provides otherwise, you may
+// not use, modify, copy, publish, distribute, disclose or transmit this
+// software or the related documents without Intel's prior written permission.
+// *****************************************************************************
```
Comment (hamishun) [inline deployments/config_psi/psi/tests/main.cpp:10]: Needs update

**Review by hamishun — CHANGES_REQUESTED** [review]

File: deployments/config_psi/scripts/README.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+
+## Introduction
+
+`encode.py` encodes data
+`decode.py` decodes data
```
Comment (hamishun) [inline deployments/config_psi/scripts/README.md:5]: We should explain how these work

**Review by hamishun — CHANGES_REQUESTED** [review]

File: deployments/config_psi/scripts/README.md (hunk 5 above)
Comment (jlhcrawford) [reply deployments/config_psi/scripts/README.md]: Description added

**Review by hamishun — APPROVED** [review]

**Pull request merged**